### DISCUSSION TITLE
Image Map coordinate discussion

### DISCUSSION THREAD

**Discussion by zhubib1234** [javalent/obsidian-leaflet#54] [Q&A]
Can coordinates only be displayed as percentages? I set the geographic coordinates, but still display the percentage coordinates.
It would be nice if it could be displayed as real geographic coordinates.

**Comment by valentine195** [comment]
Can you give me more context:

1. Is this an image map?
2. Where is it displaying them as percentages?

Regardless, they should not be displaying as percentages - the only time percentages should be used as coordinates is when initializing an image map as the initial lat/long to display on load. After that, all coordinates should be in the map's CRS.

**Reply by zhubib1234** [reply]
Don't know how to send pictures?

```leaflet
id: Map With Layered Images
image: [[18.jpg]]
image: [[merge.jpg]]
height: 400px
lat: 50
long: 50
height: 500px
minZoom: 4
maxZoom: 18
defaultZoom: 15
unit: meters
scale: 100000
```

"color":"#3accfd","layer":false,"transform":{"size":6,"x":0,"y":-2}}],"color":"#dddddd","lat":10.616478,"long":30.501693,"notePreview":false,"layerMarkers":true,"previousVersion":"3.9.5","warnedAboutMapMarker":false,"copyOnClick":true}

Is there a problem with this setting?

**Comment by valentine195** [comment]
If you drag an image into the Github comment box, it should upload it and include it into your comment.

Anyway, that all looks correct to me. Where are you then seeing coordinates as percentages?

**Reply by zhubib1234** [reply]
"command":false,"zoom":18},{"type":"default","id":"nyjhL_","loc":[-0.03313903813250363,0.030538939870893955],"link":"MarkerFile","layer":"[[18.jpg]]"

**Reply by zhubib1234** [reply]
These numbers are not geographic coordinates, are they?

**Reply by zhubib1234** [reply]
![Uploading 1.png…]()

**Reply by valentine195** [reply]
Ah, I understand your question now.

The way image maps work is different from a "real world" map, and it is related to issue #20. In order to properly use an image as a map in LeafletJS, you have to create an "overlay" that is the same dimension as the pixel dimension of the image, then project that onto the map coordinate system (which, in the case of image maps, is just a simple flat plane).

The easiest way to do this (and it is what is causing the issue I linked above) is to use the maximum zoom setting of the map to create the bounds of the map. This causes the latitude and longitude within the map to be squashed as a ratio of the dimensions of the image being displayed and the maximum zoom of the map. They do not correlate with anything geographic.

In retrospect, I regret doing it this way, and I do plan to change it. I just have to do it in a way that doesn't break people's existing markers :)

Please see:
https://github.com/valentine195/obsidian-leaflet-plugin/blob/200090fa5eb737363d52b6603cd97aeaa0326244/src/leaflet.ts#L613-L640

https://leafletjs.com/examples/crs-simple/crs-simple.html

https://stackoverflow.com/questions/13110763/is-leaflet-a-good-tool-for-non-map-images

**Reply by zhubib1234** [reply]
thank  you！

**Reply by zhubib1234** [reply]
Create an "overlay" that is the same dimension as the pixel dimension of the image. 
How to do this? Can we give an example? thank you!

**Reply by valentine195** [reply]
I was referring to the way the plugin draws the image as a map. It is technically an overlay on an empty map.

**Reply by zhubib1234** [reply]
Can we set a geographic coordinate in the upper left corner of the empty map and a geographic coordinate in the lower right corner of the map, so as to endow the empty map with geographic coordinates? In this way, I have a large amount of geocoordinate data available to import into the plug-in.

**Reply by valentine195** [reply]
Let me think on how to implement this.

**Reply by zhubib1234** [reply]
thanks！

**Reply by zhubib1234** [reply]
3.14.0 is delighted。thanks！

**Reply by zhubib1234** [reply]
How to solve the problem that the longitudinal dimension is elongated?

**Reply by valentine195** [reply]
Your bounds must be the same aspect ratio as your image, or the image will be skewed to fit.

**Reply by zhubib1234** [reply]
I don't know what's wrong. I use the corresponding geographic coordinates on the map.

**Reply by zhubib1234** [reply]
Is the aspect ratio of geographical coordinates calculated in this way?
("bottom-right-latitude"-"top-left-latitude")/("top-left-longitude"-"bottom-right-longitude")

If so
The length width ratio of geographical coordinates is “1.390266122”

The length width ratio of the picture is “9472/7936 = 1.193548387”

The two values are really different. How can we solve them?

**Reply by zhubib1234** [reply]
It's solved. I'm the one who reversed the data. thank you.
